pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java

```diff
@@ -0,0 +1,1287 @@
+ * a set of Pinot segments. The tool can be used in situations where actual
+ * source data isn't allowed to be used for the purpose of testing (regression,
+ * performance, functional, evaluation of other OLAP systems etc).
+ *
+ * The tool understands the characteristics of the given dataset (Pinot segments)
+ * and generates corresponding random data while preserving those characteristics.
+ * The tool can then also be used to generate queries for the random data.
+ *
+ * So if we have a set of production data which you want to use for testing
+ * but are unable to do so (because of security restrictions etc), then this tool
+ * can be used to generate corresponding anonymous data and queries. Users can then
+ * use the anonymized dataset (avro files) and generated queries for their testing.
+ *
+ * One avro file is generated per input Pinot segment. The tool also randomizes the
+ * column names (and table name) so that source schema is not revealed. The user is also
+ * allowed to provide a set of columns for which they want the data to be retained
+ * as is (not anonymized). User should be careful when choosing these columns. Ideally
+ * these should be time (or time related) columns since they don't reveal anything and so
+ * it is fine to copy them as is from souce segments into Avro files.
+ *
+ * Please see the implementation notes further in the code explaining the global
+ * dictionary building, and data generation and query generation phases in detail.
+ *
+ * Also, please see usage examples in
+ * {@link org.apache.pinot.tools.admin.command.AnonymizeDataCommand} to learn
+ * how this tool can be invoked from command line.
+ *
+ * Limitations:
+ * (1) Add support for multi-value columns
+ * (2) Add support for BYTES type
+ * (3) Add support for partitioning (where dataset is hash partitioned on column)
+ * (4) Add support for ORDER BY in query generator
+ * (5) Potential memory explosion for extreme high cardinality global dictionary columns
+ */
+public class PinotDataAndQueryAnonymizer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotDataAndQueryAnonymizer.class);
+
+  private final static int INT_BASE_VALUE = 1000;
+  private final static long LONG_BASE_VALUE = 100000;
+  private static final float FLOAT_BASE_VALUE = 100.23f;
+  private static final double DOUBLE_BASE_VALUE = 1000.2375;
+  private static final String DICT_FILE_EXTENSION = ".dict";
+  private static final String COLUMN_MAPPING_FILE_KEY = "columns.mapping";
+  private static final String COLUMN_MAPPING_SEPARATOR = ":";
+
+  private final String _outputDir;
+  private int _numFilesToGenerate;
+  private final String _segmentDir;
+  private final String _filePrefix;
+  // dictionaries used to generate data with same cardinality and data distribution
+  // as in source table segments
+  private final Map<String, OrigAndDerivedValueHolder> _origToDerivedValueGlobalDictionary;
```

Comment:
This seems to be a mapping between column name and value holders. Or, just the global dictionary.
Maybe call it globalDictionary, or better, columnToValuesMap ?

Reply:
Since global dictionary is maintained for per filter column, this map maps a column to its global dictionary. The global dictionary is maintained inside the OrigAndDerivedValueHolder. 
I have renamed this to columnToGlobalDictionary